test/httpErrorsReply.test.js

```diff
@@ -42,7 +36,53 @@ test('Should generate the correct http error', t => {
             message: 'I\'m a teapot',
             statusCode: 418
           })
+          // should be deleted after release of https://github.com/jshttp/http-errors/pull/73
+        } else if (code === '425') {
+          t.same(JSON.parse(res.payload), {
+            error: node14 ? 'Too Early' : 'Unordered Collection',
+            message: 'Unordered Collection',
+            statusCode: 425
+          })
+        } else {
+          t.same(JSON.parse(res.payload), {
+            error: statusCodes[code],
+            message: statusCodes[code],
+            statusCode: Number(code)
+          })
+        }
+      })
+    })
+  })
+  t.end()
+})
+
+test('Should generate the correct http error using getter', t => {
+  Object.keys(statusCodes).forEach(code => {
+    if (Number(code) < 400) return
+    t.test(code, t => {
+      t.plan(3)
+      const fastify = Fastify()
+      fastify.register(Sensible)
+
+      fastify.get('/', (req, reply) => {
+        reply.getHttpError(code)
+      })
+
+      fastify.inject({
+        method: 'GET',
+        url: '/'
+      }, (err, res) => {
+        t.error(err)
+        // should be deleted after release of https://github.com/jshttp/http-errors/pull/73
+        if ((node10 && code === '418') || (node14 && code === '425')) {
```

Comment:
I don't understand this check

right now in node14, the `res.unorderedCollection()` does not produce a 500 status code
https://runkit.com/embed/v7ceaa4vnw5y

and node 10 with `res.imateapot()` neither
https://runkit.com/embed/yzrk0ka9jedn

Reply:
in `lib/httpErrors.js` we create [statusCodesMap](https://github.com/fastify/fastify-sensible/blob/master/lib/httpErrors.js#L6) that represents mapping of HTTP status codes to normalized names. That map looks different in different versions of node.
for node 10:
```js
{ '418': 'imaTeapot', '425': 'unorderedCollection' }
```
for node 14:
```js
{'418': 'imaTeapot', '425': 'tooEarly' }
```
after that, according to values from the map, we try to access [httpErrors map](https://github.com/fastify/fastify-sensible/blob/master/lib/httpErrors.js#L19)
since we have no keys such as `imaTeapot` and `tooEarly` (for node 14+) function [getHttpError](https://github.com/fastify/fastify-sensible/blob/master/lib/httpErrors.js#L188) throws error. That is why I'm expecting errors in the test.
By the way,  we can already rename [imateapot](https://github.com/fastify/fastify-sensible/blob/master/lib/httpErrors.js#L92) method to the `imaTeapot` and remove one of the errors but it will lead to the changing of public API.

Reply:
Thank you to take the time to explain it to me 👍🏽

Reply:
you're welcome)